BE/src/main/java/team07/airbnb/controller/AccommodationController.java

```diff
@@ -3,67 +3,111 @@
+import team07.airbnb.data.accommodation.dto.request.AccommodationCreateRequest;
+import team07.airbnb.data.accommodation.dto.response.AccommodationDetailResponse;
+import team07.airbnb.data.accommodation.dto.response.AccommodationListResponse;
+import team07.airbnb.data.product.dto.response.SimpleProductResponse;
+import team07.airbnb.data.user.dto.response.TokenUserInfo;
+import team07.airbnb.data.user.enums.Role;
+import team07.airbnb.entity.AccommodationEntity;
+import team07.airbnb.service.accommodation.AccommodationService;
+import team07.airbnb.service.product.ProductService;
+import team07.airbnb.service.user.UserService;
 
+import java.time.LocalDate;
 import java.util.List;
 
+import static org.springframework.http.HttpStatus.*;
+import static team07.airbnb.data.user.enums.Role.*;
+
 @Tag(name = "숙소")
 @RequestMapping("/accommodation")
 @RestController
 @RequiredArgsConstructor
 public class AccommodationController {
     private final AccommodationService accommodationService;
+    private final ProductService productService;
     private final UserService userService;
 
+    @Tag(name = "Host")
+    @Operation(summary = "숙소 등록", description = "스쿼드비엔비에 숙소를 등록합니다.")
+    @PostMapping
+    @Authenticated(USER)
+    @ResponseStatus(OK)
+    public AccommodationListResponse createAccommodation(@RequestBody AccommodationCreateRequest createRequest, TokenUserInfo user) {
+        return AccommodationListResponse.of(accommodationService.addAccommodation(
+                createRequest.toEntity(userService.getCompleteUser(user))
+        ));
+    }
+
+    @Tag(name = "Host")
+    @Operation(summary = "숙소 삭제", description = "등록한 숙소를 삭제합니다.")
+    @DeleteMapping("/{id}")
+    @Authenticated(HOST)
+    @ResponseStatus(OK)
+    public void deleteAccommodation(@PathVariable long id, TokenUserInfo user) {
+        accommodationService.deleteById(id, userService.getCompleteUser(user));
+    }
+
+    @Tag(name = "User")
     @Operation(summary = "모든 숙소 조회", description = "스쿼드비엔비에 등록된 모든 숙소를 조회합니다.")
     @GetMapping
-    public List<AccommodationEntity> findAll() {
-        return accommodationService.findAllAccommodations();
+    @ResponseStatus(OK)
+    public List<AccommodationListResponse> findAll() {
+        return previewOf(accommodationService.findAllAccommodations());
     }
 
+    @Tag(name = "User")
+    @Operation(summary = "주변 숙소 조회", description = "지정한 위치로부터 지정한 반경 내의 숙소를 조회합니다.")
     @GetMapping("/location")
+    @ResponseStatus(OK)
     public List<AccommodationListResponse> findNeighbor(
             @RequestParam double longitude,
             @RequestParam double latitude,
             @RequestParam double distance) {
 
-        return accommodationService.findNearbyAccommodations(longitude, latitude, distance)
-                .stream()
-                .map(AccommodationListResponse::of)
-                .toList();
+        return previewOf(accommodationService.findNearbyAccommodations(longitude, latitude, distance * 1000));
```

Comment:
스타일이긴 한데요
```suggestion
        return previewOf(
          accommodationService.findNearbyAccommodations(
            longitude, 
            latitude, 
            distance * 1000
          )
        );
```
는 어떨까요? 공백을 포함하여 총 110자 인 라인인데, 120자 부터는 라인개행 강력추천이고 지금정도면 권고 정도 일꺼같아요~!